crates/orchestrator/src/queue/mod.rs

```diff
@@ -6,11 +6,87 @@ use std::time::Duration;
 
 use async_trait::async_trait;
 use color_eyre::Result as EyreResult;
+use lazy_static::lazy_static;
 use mockall::automock;
 use omniqueue::{Delivery, QueueError};
 
 use crate::config::Config;
 use crate::jobs::JobError;
+use crate::setup::SetupConfig;
+
+#[derive(Clone)]
+pub struct QueueConfig {
+    pub name: String,
+    pub visibility_timeout: i32,
+    pub max_receive_count: Option<i32>,
+    pub dlq_name: Option<String>,
```

Comment:
is it better to use &str?

Reply:
changed to &str

Reply:
If I use string then we need to specify lifetime here. Ig we can move forward with String.